analyses/proteomic-data/01-add-bsid-to-proteomic-data.R

```diff
@@ -0,0 +1,76 @@
+library(tidyverse)
+
+## set directoies
+root_dir <- rprojroot::find_root(rprojroot::has_dir(".git"))
+data_dir <- file.path(root_dir, "data")
+input_dir <- file.path(root_dir, "analyses", "proteomic-data", "input")
+output_dir <- file.path(root_dir, "analyses", "proteomic-data", "output")
+if(!dir.exists(output_dir)){
+  dir.create(output_dir)
+}
+
+## read in files
+hist <- readr::read_tsv(file.path(data_dir, "Hope-GBM-histologies-base.tsv"))
+whole_cell_protein <- data.table::fread(file.path(input_dir, "harmonized_proteome_imputed_data_061623.tsv"))
+pho <- data.table::fread(file.path(input_dir, "harmonized_phosphosite_imputed_data_ischemia_removed_062823.tsv"))
+pho_motif <- data.table::fread(file.path(input_dir, "harmonized_phosphosite_imputed_data_ischemia_removed_motif_070323.tsv"))
+
+# Add bs_id to whole cell proteomic
+whole_cell_id <- data.frame(sample_id = colnames(whole_cell_protein)[-c(1:3)]) %>% 
+  left_join(hist) %>% 
+  filter(experimental_strategy == "Whole Cell Proteomics" | grepl("^C3", sample_id)) %>%
+  select(Kids_First_Biospecimen_ID, sample_id) %>% 
+  arrange(sample_id)
+
+## generate a named vector with sample_id and bs_id
+col_mapping <- setNames(whole_cell_id$Kids_First_Biospecimen_ID, whole_cell_id$sample_id)
+
+## assign bs_id to whole cell proteomic matrix
+colnames(whole_cell_protein) <- c("ApprovedGeneSymbol", "OldSymbol", "coding_gene", 
```

Comment:
Instead of hardcoding the three column names, can consider using colnames(whole_cell_protein)[1:3]. This would match the strategy used in similar lines below (line 47, line 71)

Reply:
Thank you! I made changes accordingly.